src/main/java/net/openhft/chronicle/threads/MediumEventLoop.java

```diff
@@ -230,17 +226,18 @@ private boolean addHandlerBeforeStart(@NotNull EventHandler handler) {
      * This is the code executed when a non-event-loop thread wants to add a handler on a started loop
      */
     private void addHandlerAfterStart(@NotNull EventHandler handler) {
-        do {
-            if (isStopped()) {
-                if (Jvm.isDebugEnabled(MediumEventLoop.class)) {
-                    Jvm.debug().on(MediumEventLoop.class, "Aborted adding handler because event loop was stopped, handler=" + handler);
-                }
-                return;
+        if (isStopped()) {
+            if (Jvm.isDebugEnabled(MediumEventLoop.class)) {
+                Jvm.debug().on(MediumEventLoop.class, "Aborted adding handler because event loop was stopped, handler=" + handler);
             }
-            pauser.unpause();
+            return;
+        }
+
+        pauser.unpause();
```

Comment:
I assume this is to prompt the event loop to pick up the new handler if it's pausing? we should move it to after we've offered the handler.